apps/music/js/ui/title_bar.js

```diff
@@ -97,25 +97,28 @@ var TitleBar = {
               PlayerView.currentIndex
             ];
             var playingBlob = PlayerView.playingBlob;
-            AlbumArtCache.getCoverBlob(currentFileinfo).then(function(picture) {
-              var currentMetadata = currentFileinfo.metadata;
-              App.pendingPick.postResult({
-                type: playingBlob.type,
-                blob: playingBlob,
-                name: currentMetadata.title || '',
-                // We only pass some metadata attributes so we don't share
-                // personal details like # of times played and ratings.
-                metadata: {
-                  title: currentMetadata.title,
-                  artist: currentMetadata.artist,
-                  album: currentMetadata.album,
-                  picture: picture
-                }
-              });
-
-              cleanupPick();
-            });
 
+            LazyLoader.load('js/metadata/album_art_cache.js', function() {
+              AlbumArtCache.getThumbnailBlob(currentFileinfo).then(
+                function(picture) {
+                  var currentMetadata = currentFileinfo.metadata;
+                  App.pendingPick.postResult({
+                    type: playingBlob.type,
+                    blob: playingBlob,
+                    name: currentMetadata.title || '',
+                    // We only pass some metadata attributes so we don't share
+                    // personal details like # of times played and ratings.
+                    metadata: {
+                      title: currentMetadata.title,
+                      artist: currentMetadata.artist,
+                      album: currentMetadata.album,
+                      picture: picture
+                    }
+                  });
+
+                  cleanupPick();
+                });
+            });
```

Comment:
Looks like all you're doing here is adding a LazyLoader.load() wrapper.  Too bad that function doesn't have a cool name like `using()` or something.

I'll just add that this seems like an obvious refactoring opportunity to (in some future bug) move pick-related functionality out of the title bar view!